py/strato/common/log/log2text.py

```diff
@@ -227,20 +227,79 @@ def printLogs(logFiles, formatter):
 
 
 def updateConfFile(field, value):
-    with open(LOG_CONFIG_FILE_PATH, 'r') as f:
-        conf = yaml.load(f.read())
+    try:
+        with open(LOG_CONFIG_FILE_PATH, 'r') as f:
+            conf = yaml.load(f.read())
+    except:
+        print "No configuration file was found. creating new"
+        conf = {}
+    finally:
         conf[field] = value
     with open('/tmp/stratolog-conf.tmp', 'w') as f:
         f.write(yaml.dump(conf))
+        f.close()
     os.system('sudo mv /tmp/stratolog-conf.tmp %s' % LOG_CONFIG_FILE_PATH)
 
 
-def universalOpen(filePath, mode='r'):
+def universalOpen(filePath, mode='r', tail=0):
     if filePath.endswith('.gz'):
-        return gzip.GzipFile(filePath, mode)
+        fileObj = gzip.GzipFile(filePath, mode)
     else:
-        return open(filePath, mode)
-
+        fileObj = open(filePath, mode)
+    if tail > 0:
+        tailFile(fileObj, tail)
+    return fileObj
+
+def tailFile(fileObj, n):
+    bufferSize = bufferIncrement = 8192
+    fileObj.seek(0,2)
+    fileSize = fileObj.tell()
+    while bufferSize < fileSize:
+        fileObj.seek(-bufferSize,2)
+        linesInBuffer = len([line for line in fileObj.readlines() if line.strip()])
+        if linesInBuffer > n:
+            fileObj.seek(-bufferSize,2)
+            while linesInBuffer != n:
+                fileObj.readline()
+                linesInBuffer -= 1
+            return fileObj
+        else:
+            bufferSize += bufferIncrement
+    fileObj.seek(0, 0)
+    return fileObj
+
+def ping(host):
+    return os.system("ping -c 1 %s > /dev/null 2>&1" % host) == 0
+
+def runRemotely(host, ignoreArgs):
+    try:
```

Comment:
This is strange flow - you already got the args, so why do you parse the file if not needed? And you already opened it in init. So why to read it again?
Its really hard to follow.
1. Check if should run remotely.